### PR TITLE
[pre-adoption-validation]Check kernel args

### PR DESCRIPTION
**Pull request by gibizer**
The edpm_kernel role checks is the content of the edpm_kernel_args
is already in the /proc/cmdline and if it isn't then it reconfigure grub
and request a reboot. This pre-adoption check therefore checks the same
and fails the adoption early to avoid kernel arg reconfiguration during
the edpm_kernel role run.

The edpm_kernel_hugepages ansible var can be used independently from the
edpm_kernel_args var to pass huge page kernel config. So the kernel args
pre-adoption check is extended to consider this variable as well.

Implements: [OSPRH-5714](https://issues.redhat.com//browse/OSPRH-5714)

### CODE REVIEW

**Comment by gibizer** [comment]
/hold
Lets have a test run of this in data-plane-adoption...

File: roles/edpm_pre_adoption_validation/molecule/kernel-args-existing-args/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  name: ansible
+  inventory:
+    hosts:
+      all:
+        vars:
+          edpm_pre_adoption_validation_hostname_enabled: false
+          # this is a kernel arg that by default defined in the test env
+          # we use this to simulate that what is requested during adoption
+          # is already configured
+          edpm_kernel_args: "net.ifnames=0"
```
Comment (gibizer) [inline roles/edpm_pre_adoption_validation/molecule/kernel-args-existing-args/molecule.yml:25]: and it turned out that the github runner uses different images with different cmdline than my local env.

**Review by bogdando — COMMENTED** [review]
lgtm

**Comment by gibizer** [comment]
the data-plane adoption CI run the new check here:

https://logserver.rdoproject.org/21/421/8c69005e4470ad2a98dfea3bee5292f13346fc9f/github-check/data-plane-adoption-osp-17-to-extracted-crc-minimal-no-ceph/1d2652d/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/pods/pre-adoption-validation-openstack-pre-adoption-openstack-dsqnz/logs/openstackansibleee.log

File: roles/edpm_pre_adoption_validation/defaults/main.yml [hunk 2]
```diff
@@ -18,3 +18,6 @@
 # All variables intended for modification should be placed in this file.
 edpm_pre_adoption_validation_old_nova_config: /var/lib/config-data/puppet-generated/nova_libvirt/etc/nova/nova.conf
 edpm_pre_adoption_validation_old_neutron_config: /var/lib/config-data/puppet-generated/neutron/etc/neutron/neutron.conf
+
+edpm_pre_adoption_validation_hostname_enabled: true
+edpm_pre_adoption_validation_kernel_args_enabled: true
```
Comment (jistr) [inline roles/edpm_pre_adoption_validation/defaults/main.yml:23]: :+1:

File: roles/edpm_pre_adoption_validation/tasks/main.yml [hunk 3]
```diff
@@ -14,55 +14,11 @@
...
-          "import configparser as c;
-          p = c.ConfigParser(strict=False);
-          p.read('{{ edpm_pre_adoption_validation_old_nova_config }}');
-          print(p['DEFAULT']['host'])"
-      register: nova_old_config_output
-      changed_when: false
-      failed_when: nova_old_config_output.stdout != canonical_hostname
+- name: Validate hostname
+  when: edpm_pre_adoption_validation_hostname_enabled
+  ansible.builtin.include_tasks: hostname.yml
```
Comment (jistr) [inline roles/edpm_pre_adoption_validation/tasks/main.yml:20]: :+1:

**Review by jistr — COMMENTED** [review]
Nice, thank you!

**Pull request merged**